jobs/azure-discovery/src/main/java/com/tmobile/pacbot/azure/inventory/file/S3Uploader.java

```diff
@@ -0,0 +1,193 @@
+		log.info("Uploading files to bucket: {} folder: {}",s3Bucket,dataFolderS3);
+		TransferManager xferMgr = TransferManagerBuilder.standard().withS3Client(s3client).build();
+		try {
+		    MultipleFileUpload xfer = xferMgr.uploadDirectory(s3Bucket,
+		    		dataFolderS3, new File(filePath), false);
+		   
+		    while(!xfer.isDone()){
+		    	try{
+		    		Thread.sleep(3000);
+		    	}catch(InterruptedException e){
+		    	    log.error("Error in uploadAllFiles",e);
+		    	    ErrorManageUtil.uploadError("all", "all", "all", e.getMessage());
+		    	    Thread.currentThread().interrupt();
+		    	}
+		    	log.debug("    Transfer % Completed :" +xfer.getProgress().getPercentTransferred());
+		    }
+		    xfer.waitForCompletion();
+		   
+		   log.info("Transfer completed");
+		} catch (Exception e) {
+			log.error("{\"errcode\": \"S3_UPLOAD_ERR\" ,\"account\": \"ANY\",\"Message\": \"Exception in loading files to S3\", \"cause\":\"" +e.getMessage()+"\"}") ;
+			ErrorManageUtil.uploadError("all", "all", "all", e.getMessage());
+		}
+		 xferMgr.shutdownNow();
+	}
+	
+	/**
+	 * Copyto back up.
+	 *
+	 * @param s3client the s 3 client
+	 * @param s3Bucket the s 3 bucket
+	 * @param from the from
+	 * @param to the to
+	 */
+	private void copytoBackUp(AmazonS3 s3client,String s3Bucket,String from, String to){
+		String[] keys = listKeys(s3client,s3Bucket,from);
+		String fileName ="";
+		for(String key:keys){
+			try{
+				fileName = key.substring(key.lastIndexOf('/')+1);
+				s3client.copyObject(s3Bucket,key,s3Bucket,to+"/"+fileName);
+				log.debug("    Copy "+fileName + " to backup folder");
+			}catch(Exception e){
+				log.info("    Copy "+fileName + "failed",e);
+				ErrorManageUtil.uploadError("all", "all", "all", e.getMessage());
+			}
+		}
+	}
+	
+	/**
+	 * Delete files.
+	 *
+	 * @param s3client the s 3 client
+	 * @param s3Bucket the s 3 bucket
+	 * @param folder the folder
+	 */
+	private void deleteFiles(AmazonS3 s3client,String s3Bucket,String folder){
+		
+		String[] keys = listKeys(s3client,s3Bucket,folder);
+		DeleteObjectsRequest multiObjectDeleteRequest = new DeleteObjectsRequest(s3Bucket).withKeys((keys));
+		
+		try{
+			DeleteObjectsResult result = s3client.deleteObjects(multiObjectDeleteRequest);
+			log.debug("Files Deleted " +result.getDeletedObjects().stream().map(obj->obj.getKey()).collect(Collectors.toList()));
+		}catch(Exception e){
+			log.error("Delete Failed",e);
+			ErrorManageUtil.uploadError("all", "all", "all", e.getMessage());
+		}
+	}
+	
+	/**
+	 * List keys.
+	 *
+	 * @param s3client the s 3 client
+	 * @param s3Bucket the s 3 bucket
+	 * @param folder the folder
+	 * @return the string[]
+	 */
+	private String[] listKeys(AmazonS3 s3client,String s3Bucket,String folder){
+		try{
+			return s3client.listObjectsV2(new ListObjectsV2Request().withBucketName(s3Bucket).withPrefix(folder)).getObjectSummaries().stream().map(S3ObjectSummary::getKey).toArray(String[]::new);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

This code might not produce accurate results if the operation returns paginated results instead of all results. Consider adding another call to check for additional results.